src/test/java/org/elasticsearch/test/disruption/LongGCDisruption.java

```diff
@@ -23,76 +24,50 @@
 import java.util.HashSet;
 import java.util.Random;
 import java.util.Set;
-import java.util.concurrent.atomic.AtomicInteger;
 import java.util.regex.Pattern;
 
+/**
+ * Suspends all threads on the specified node in order to simulate a long gc.
+ */
 public class LongGCDisruption extends SingleNodeDisruption {
 
-    volatile boolean disrupting;
-    volatile Thread worker;
-
-    final long intervalBetweenDelaysMin;
-    final long intervalBetweenDelaysMax;
-    final long delayDurationMin;
-    final long delayDurationMax;
-
-
-    public LongGCDisruption(Random random) {
-        this(null, random);
-    }
-
-    public LongGCDisruption(String disruptedNode, Random random) {
-        this(disruptedNode, random, 100, 200, 300, 20000);
-    }
+    private final static Pattern[] unsafeClasses = new Pattern[]{
+            // logging has shared JVM locks - we may suspend a thread and block other nodes from doing their thing
+            Pattern.compile("Logger")
+    };
 
-    public LongGCDisruption(String disruptedNode, Random random, long intervalBetweenDelaysMin,
-                            long intervalBetweenDelaysMax, long delayDurationMin, long delayDurationMax) {
-        this(random, intervalBetweenDelaysMin, intervalBetweenDelaysMax, delayDurationMin, delayDurationMax);
-        this.disruptedNode = disruptedNode;
-    }
+    protected final String disruptedNode;
+    private volatile Set<Thread> suspendedThreads;
```

Comment:
strictly speaking, this doesn't need to be volatile. We update it under a lock which guarantees the visibility of the changes.